vmware/kubernetes
# VSAN policy support in Kubernetes

Following test cases have been under testing:

Test cases:

1. Validation of VSAN capabilities.
- hostFailuresToTolerate : Minimum 1 and Max 3. Should be integer.
- forceProvisioning: Should be either 0 or 1 . Should be integer.
- cacheReservation: Expressed in percentage. Should be between 0 and 100. Should be integer.
- stripeWidth: Minimum is 1 and Maximum is 12. Should be integer.
- proportionalCapacity: Expressed in percentage. Should be between 0 and 100. Should be integer.
- iopsLimit: Should be greater than 0. Should be integer.

2. Use a VSAN testbed setup. Try valid VSAN capabilities which are supported by VSAN testbed. Make sure the disk is created with policy configured with it.
- Ex: Using hostFailuresToTolerate=1 and cacheReservation=12

3. Use a VSAN testbed setup. Try valid VSAN capabilities which are not supported by VSAN testbed. Make sure that the disk creation fails and PVC status is pending.

4. Use a non-VSAN testbed setup. Try using VSAN capabilities on a non-VSAN testbed. PVC status will be pending and it errors to the user saying to VSAN capabilities are not supported on a non-VSAN testbed.

5. Use a non-VSAN testbed setup. Use a normal storage class YAML without VSAN capabilities. The disk should be created successfully.

6. Try all 1 to 5 with custom datastore specified by the user.

7. Try thick, thin and eagerzeroedthick provisioning of disks. Since the code is changed for this module, this has to be tested.
